apps/microsoft-teams/frontend/src/components/config/ContentTypeSelectionModal/ContentTypeSelectionModal.tsx

```diff
@@ -30,11 +31,15 @@ const ContentTypeSelectionModal = (props: Props) => {
     error,
   } = props;
 
+  const [searchQuery, setSearchQuery] = useState<string>('');
   const [selectedContentTypeId, setSelectedContentTypeId] = useState(savedContentTypeId ?? '');
-
-  const { title, button, link, emptyContent, emptyHeading, errorMessage } =
+  const { title, button, link, emptyContent, emptyHeading, errorMessage, searchPlaceholder } =
     contentTypeSelection.modal;
 
+  const handleSearchQueryUpdate = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
probably worth adding a comment explaining how this component is brokering the search query between `<DeboundedSearchInput>` and `<ResultsList>` and why it is necessary: updating state of either of those 2 components should not force the other to rerender, so we needed to push some state down to child components.